src/select-list-view.ts

```diff
@@ -1,9 +1,30 @@
-const {Disposable, CompositeDisposable, TextEditor} = require('atom')
+// TODO: fix redundant returns in methods
+
 const etch = require('etch')
+import { Disposable, CompositeDisposable, TextEditor, CommandEvent } from 'atom'
 const $ = etch.dom
-const fuzzaldrin = require('fuzzaldrin')
+import fuzzaldrin from 'fuzzaldrin'
+
+// TODO: etch types
+export type EtchElement = HTMLElement
+
+import { SelectListProperties } from './select-list-properties'
+
+export default class SelectListView {
+  /** When creating a new instance of a select list, or when calling `update` on an existing one, you can supply an object with the typeof SelectListProperties */
+  props: SelectListProperties
+
+  /** an array containing the objects you want to show in the select list. */
+  items: Array<object | string> // TODO: Added initializer! Either fix this.items or assign it in constructor
```

Comment:
Can you elaborate on this comment (and the identical one in the constructor)?

Reply:
`items` is identical in `props` and the constructor. So, for backward compatibility, I did not remove it. But it can be removed in a major version bump.

https://github.com/aminya/atom-select-list/blob/af6f304c7ae6072e8d253f0da918f6801a05b84d/src/select-list-view.ts#L39

Reply:
I looked through the code to re-familiarize myself with this. It technically is assigned in the constructor, via `computeItems`. Does Typescript throw an error if it's not explicitly set in the constructor itself? I forget how good it is at following the codeflow.

I'm not positive, but I think it is necessary because of the slices we do on it; this way `this.props.items` is the original copy and `this.items` is the filtered & ordered list that's shown.

Reply:
And: is this meant to be public?